pkg/virt-controller/watch/vmi.go

```diff
@@ -513,6 +517,10 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 	return nil
 }
 
+func isTempPodFailed(pod *k8sv1.Pod) bool {
+	return pod.Status.Phase == k8sv1.PodFailed || pod.DeletionTimestamp != nil
```

Comment:
@davidvossel, @mhenriks Here I check for failing temp Pod. Do you think there are other scenarios than simple Failed state, or Unschedulable pod (I think I need to check for this unschedulable state)? An what else I should check here?

Basically if a pod is successfully scheduled then the PVCs should be bound. I am trying to think about other pathological cases to cover here, so we do not miss any problems.

Reply:
I have now added check for pod condition PodScheduled with Reason == k8sv1.PodReasonUnschedulable, and mark VMI as failed. 
Now when I test this I can see that VMI fails and is retried by controller. Is it good behavior? VM shows this:
```
Events:
  Type    Reason                      Age                   From                       Message
  ----    ------                      ----                  ----                       -------
  Normal  SuccessfulDataVolumeCreate  23m                   virtualmachine-controller  Created DataVolume alpine-dv
  Normal  SuccessfulDelete            21m                   virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance 871a8973-0f2e-4744-852f-cfeb85c36fc3
  Normal  SuccessfulDelete            21m                   virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance d62d5c34-5dc6-46df-924d-566e0ab20eb3
  Normal  SuccessfulDelete            21m                   virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance 967b46fa-0022-4d1f-b431-a6304fc96ab2
  Normal  SuccessfulDelete            21m                   virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance 11940934-ae44-4a36-8276-1f10ab862f41
  Normal  SuccessfulDelete            21m                   virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance 0ff178fd-e979-402f-84bb-6b8539f8ae4f
  Normal  SuccessfulDelete            21m                   virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance 5e0d40d9-b4fe-45fe-9523-b97558afbd72
  Normal  SuccessfulDelete            21m                   virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance a7e706e8-559f-4a15-8dcc-0fe876bfe832
  Normal  SuccessfulDelete            21m                   virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance 4e79a8c0-b699-4298-a140-d13b9e3e3323
  Normal  SuccessfulDelete            21m                   virtualmachine-controller  Stopped the virtual machine by deleting the virtual machine instance 34c3dcb8-8e63-4135-a860-89974c046ce5
  Normal  SuccessfulCreate            21m (x12 over 21m)    virtualmachine-controller  Started the virtual machine by creating the new virtual machine instance vm-alpine-datavolume
  Normal  SuccessfulDelete            3m6s (x564 over 21m)  virtualmachine-controller  (combined from similar events): Stopped the virtual machine by deleting the virtual machine instance 5e424c0f-3176-4a7a-9394-3fa6d4f5714e

```

Reply:
I think the pod will just be pending forever when can't schedule so not sure you can deal with that without maybe looking at events.  Should be easy enough to test?

Reply:
looks like you might have resolved this?

Reply:
looks like it is by design, ` runStrategy: "RerunOnFailure` vs `runStrategy: Manual`